src/main/java/com/google/sps/data/GetGovAdaptor.java

```diff
@@ -8,22 +8,24 @@
 import java.io.IOException;
 import java.io.OutputStream;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 import com.google.gson.Gson;
 import com.google.gson.reflect.TypeToken;
 
 /** File to run the GET request in java before putting it in the main adaptor function. 
  */
 public class GetGovAdaptor {
-  public static void main(final String[] args) {
-    try {
-      GetGovAdaptor.getSiteInfo();
-    } catch (final Exception e) {
-      System.out.println(e.getMessage());
-    }
+
+  public static Coordinates getCoord(HashMap<Integer, GovSiteDetails> Map, Integer siteId) {
+    return new Coordinates(Map.get(siteId).lng, Map.get(siteId).lat);
   }
 
-  public static ArrayList<NSWGovAQDataPoint> getSiteInfo() throws Exception {
+  public static HashMap<Integer, GovSiteDetails> getSiteInfo() throws Exception {
```

Comment:
Make this 'private', and have it called by a default constructor to populate the map automatically when the class is created.
Optionally (and fine to be left for later) have getCoord() re-use it if it fails to look up a siteId in the stored map.